cardano-node/src/Cardano/Node/Tracing/API.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MonoLocalBinds #-}
+{-# LANGUAGE PackageImports #-}
+{-# LANGUAGE RankNTypes #-}
+module Cardano.Node.Tracing.API
+  (initTraceDispatcher)
+where
+
+import           Prelude
+
+import           "contra-tracer" Control.Tracer (traceWith)
+import           Data.Maybe (fromMaybe)
+import           Data.Time.Clock (getCurrentTime)
+
+import           System.Metrics as EKG
+
+import           Network.Mux.Trace (TraceLabelPeer (..))
+
+import           Ouroboros.Consensus.Ledger.Inspect (LedgerEvent)
+import           Ouroboros.Consensus.MiniProtocol.ChainSync.Client (TraceChainSyncClientEvent)
+import           Ouroboros.Consensus.Node (NetworkP2PMode, RunNode)
+import           Ouroboros.Network.ConnectionId (ConnectionId)
+import           Ouroboros.Network.Magic (NetworkMagic)
+import           Ouroboros.Network.NodeToClient (withIOManager)
+import           Ouroboros.Network.NodeToNode (RemoteAddress)
+
+import           Cardano.Node.Configuration.POM (NodeConfiguration (..), ncProtocol)
+import           Cardano.Node.Protocol.Types
+import           Cardano.Node.Queries
+import           Cardano.Node.Startup
+import           Cardano.Node.TraceConstraints
+import           Cardano.Node.Tracing
+import           Cardano.Node.Types
+
+import           Cardano.Logging hiding (traceWith)
+import           Cardano.Node.Tracing.Tracers
+import           Cardano.Node.Tracing.Tracers.Peer (startPeerTracer)
+import           Cardano.Node.Tracing.Tracers.Resources (startResourceTracer)
+
+initTraceDispatcher ::
```

Comment:
This doesn't need a separate module. Let's move it to `Cardano.Node.Tracing`.

Reply:
Wouldn't like to do, as Cardano.Node.Tracing is already quite big

Reply:
Ok, will try this

Reply:
The problem here is that the individual tracers reference the data type in Cardano.Node.Tracing, and when I move the procedure to this module I get cyclic dependencies, cause the function references individual tracers. So I would suggest to stay with the current structure.